langwatch/src/server/api/routers/onboarding.ts

```diff
@@ -0,0 +1,107 @@
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { skipPermissionCheck } from "../permission";
+import { dependencies } from "../../../injection/dependencies.server";
+import * as Sentry from "@sentry/nextjs";
+import { organizationRouter } from "./organization";
+import { projectRouter } from "./project";
+
+/**
+ * Input schema for organization signup data
+ */
+const signUpDataSchema = z.object({
+  usage: z.string().optional().nullable(),
+  solution: z.string().optional().nullable(),
+  terms: z.boolean().optional(),
+  companyType: z.string().optional().nullable(),
+  companySize: z.string().optional().nullable(),
+  projectType: z.string().optional().nullable(),
+  howDidYouHearAboutUs: z.string().optional().nullable(),
+  otherCompanyType: z.string().optional().nullable(),
+  otherProjectType: z.string().optional().nullable(),
+  otherHowDidYouHearAboutUs: z.string().optional().nullable(),
+  utmCampaign: z.string().optional().nullable(),
+});
+
+/**
+ * Router for handling onboarding-related operations.
+ */
+export const onboardingRouter = createTRPCRouter({
```

Comment:
We'd discussed breaking this up into smaller functions, SRP. I know we ran into some trouble with the types for the function interface, but it's probably still worth doing.